app/assets/javascripts/verification_interface.js

```diff
@@ -0,0 +1,146 @@
+});
+
+$(document).on("ready turbolinks:load", function() {
+  handleActionTable();
+});
+
+function handleAttachAndSubmit() {
+  // Oversees file attach/remove and submission
+  class FileManager {
+    constructor(uploadFormId, fileInputId, fileListElementId, uploadButtonId) {
+      this.attachedFiles = [];
+      this.uploadForm = document.getElementById(uploadFormId);
+      this.fileInput = document.getElementById(fileInputId);
+      this.fileListElement = document.getElementById(fileListElementId);
+      this.uploadButton = document.getElementById(uploadButtonId);
+    }
+
+    /**
+     * Setup event listeners for the file input to handle attachments and form submission to upload the attached files.
+     */
+    setupListeners() {
+      if (!this.uploadForm || !this.fileInput || !this.fileListElement || !this.uploadButton) { return }
+
+      // Reload the file list when files are attached
+      this.fileInput.addEventListener('change', (event) => {
+        this.attachedFiles = this.attachedFiles.concat(Array.from(this.fileInput.files));
+        this.updateFileList();
+        this.fileInput.value = ''; // Reset the file input value to allow re-selecting the same file
+      });
+
+      // Override submission to send the attached files as form data by updating the `fileInput` with `DataTransfer`.
+      this.uploadForm.addEventListener('submit', (event) => {
+        event.preventDefault();
+  
+        // Create a DataTransfer object to hold the files
+        const dataTransfer = new DataTransfer();
+        this.attachedFiles.forEach(file => {
+          dataTransfer.items.add(file);
+        });
+    
+        // Set the files to the file input
+        this.fileInput.files = dataTransfer.files;
+    
+        // Submit the form
+        this.uploadForm.submit();
+      });
+    }
+  
+    /**
+     * Reload the `fileListElement` with the `attachedFiles`, building out the file list with `AttachFileElement` instances.
+     * If no files are selected, display a message indicating that no files are selected and disable the upload button.
+     */
+    updateFileList() {
+      this.fileListElement.innerHTML = ''; // Clear the current file list
+  
+      if (this.attachedFiles.length) {
+        const fileItems = this.attachedFiles.map((file) => new this.AttachFileElement(file, this));
+        fileItems.forEach((fileItem) => { this.fileListElement.appendChild(fileItem.element) });
+      } else {
+        this.fileListElement.innerHTML = `<li>none</li>`;
+      }
+      this.toggleUploadButton()
+    }
+  
+    /**
+     * Enable or disable the upload button based on whether files are attached.
+     */
+    toggleUploadButton() {
+      if (this.attachedFiles.length) {
+        this.uploadButton.removeAttribute('disabled');
+      } else {
+        this.uploadButton.setAttribute('disabled', 'disabled');
+      }
+    }
+  
+    // Repesents a file attached to the form
+    AttachFileElement = class {
```

Comment:
I'm open to introducing a partial for this instead, so when we need to render these attachment rows after the user uses the input, we can just render the partial instead of constructing the element in JS as we are here.

The con is that it's going to be a super short partial only ever used here...